workflow/controller/operator.go

```diff
@@ -1422,9 +1433,17 @@ func (woc *wfOperationCtx) markWorkflowPhase(phase wfv1.NodePhase, markCompleted
 				woc.wf.ObjectMeta.Labels = make(map[string]string)
 			}
 			woc.wf.ObjectMeta.Labels[common.LabelKeyCompleted] = "true"
-			err := woc.controller.wfArchive.ArchiveWorkflow(woc.wf)
+			err := woc.deletePDBResource()
+			if err != nil {
+				woc.wf.Status.Phase = wfv1.NodeError
```

Comment:
this looks wrong to me, isn't there a. woc.markErrorPhase or similar?

Reply:
`woc.markWorkflowError` is calling again to `woc.markWorkflowPhase`.  I thought of extract `setphaselabelandmessage as function but it looks overengineering for just 4 statements.